### DISCUSSION TITLE
x-data not reactive when pulling from store

### DISCUSSION THREAD

**Discussion by brandoncarl** [alpinejs/alpine#3590] [1. Help]
Hello – I am wondering if there is an easier way to do the following?

```html
<div id="sidebar-list" x-data="$store.library.artists">
  <template x-for="artist in $store.library.artists">
    <a x-text="artist" @click="$store.state.setArtist(artist)"></a>
  </template>
</div>
```

I'd like to do this, but when I do, I lose reactivity. It's unintuitive to me as to why:
```html
<div id="sidebar-list" x-data="{ artists: $store.library.artists }">
  <template x-for="artist in artists">
    <a x-text="artist" @click="$store.state.setArtist(artist)"></a>
  </template>
</div>
```

Thanks in advance for the help

**Comment by ekwoka** [comment]
This should still be reactive no problem.

What is the action that `setArtist` does? I guess it must replace the whole array? Yeah, that wouldn't be reactive, becuase the data isn't whatever the current array is, but what the array was then. So either don't replace the array, or change the `artists` to a getter.

```html
x-data="{
  get artists() { return this.$store.library.artists }
  }"
  ```

Generally issues with `x-for` are caused by not providing a `:key` which is something you should ALWAYS do :)

**Reply by brandoncarl** [reply]
Thank you, will try this.
